agent/doctor/doctor.go

```diff
@@ -54,19 +68,49 @@ func (doc *Doctor) RunHealthchecks() bool {
 	return doc.allRight(allChecksResult)
 }
 
+// GetPublishInstanceStatusRequest will get all healthcheck statuses and generate
+// a sendable PublishInstanceStatusRequest
+func (doc *Doctor) GetPublishInstanceStatusRequest() (*ecstcs.PublishInstanceStatusRequest, error) {
+	metadata := &ecstcs.InstanceStatusMetadata{
+		Cluster:           aws.String(doc.cluster),
+		ContainerInstance: aws.String(doc.containerInstanceArn),
+		RequestId:         aws.String(uuid.NewRandom().String()),
+	}
+	instanceStatuses := doc.getInstanceStatuses()
+
+	if instanceStatuses != nil {
+		return &ecstcs.PublishInstanceStatusRequest{
+			Metadata:  metadata,
+			Statuses:  instanceStatuses,
+			Timestamp: aws.Time(time.Now()),
+		}, nil
+	} else {
+		return nil, EmptyHealthcheckError
```

Comment:
Will an error be returned when Agent just started and there are no health status to send ?

Reply:
update: `RunHealthchecks` will always run all the available healthchecks.  If any of the checks fails for any reason, the impaired status will reflect the failure.  The return value is the overall health: 
`return doc.allRight(allChecksResult)`

So if we don't have any instance statuses, then this indicates that the healthcheck(s) has/have not been run for some reason and it should result in an error as we always precede the message generation with `RunHealthchecks()`

After the first run of the healthchecks, the worst case will be that we end up sending messages with stale/duplicate status data (including timestamp).  In the message sending changes, we should add handling for this case -- marking a specific check as `sent` once its message has been passed on to TACS.

Reply:
Yes they should never be empty and we should not see this error ever.